js/common/modules/@arangodb/aql/explainer.js

```diff
@@ -2135,6 +2135,11 @@ function profileQuery(data, shouldPrint) {
   stringBuilder.clearOutput();
   let stmt = db._createStatement(data);
   let cursor = stmt.execute();
+  if (options.stream) {
+    while(cursor.hasNext()) {
+      cursor.next();
+    }
```

Comment:
Imagine this running in arangosh... This will do a lot of back-and-forth communication with arangod. No doubt it will work, but this is very likely inefficient.

Reply:
Well, it fetches all batches. But I think that's the only sensible thing to do in order to profile a streaming query. At least I currently can't think of something better.

Reply:
We could do this server-sided instead of client-sided.

Reply:
My thought was that making this server-sided will be effectively the same as setting `stream: false`, rendering it somewhat useless. But perhaps the query profiling doesn't make much sense with a streaming cursor, and we should just forbid it, or emit a warning. I'm fine either way.

Reply:
I will try to add a warning here. 
Apart from that, I think we can leave the change in this PR as it is, i.e. merge it. 
Let me add that warning and a test now.

Reply:
Changed my mind and fixed it properly. See latest commit.